cobra/test/test_flux_analysis/test_find_active_reactions.py

```diff
@@ -0,0 +1,94 @@
+""" Test functionalities of find_active_reactions"""
+
+
+import pytest
+
+import cobra
+from cobra.test import create_test_model
+from cobra.flux_analysis import (
+    find_active_reactions, find_reactions_in_cycles, fastSNP)
+
+
+def test_find_active_reactions_benchmark(model, benchmark, all_solvers):
+    """Benchmark find_active_reactions."""
+
+    model.solver = all_solvers
+    benchmark(find_active_reactions, model)
+
+
+def test_find_reactions_in_cycles_benchmark(model, benchmark, all_solvers):
+    """Benchmark find_reactions_in_cycles."""
+
+    model.solver = all_solvers
+    benchmark(find_reactions_in_cycles, model)
+
+
+def test_fastSNP_benchmark(model, benchmark, all_solvers):
+    """Benchmark fastSNP."""
+
+    model.solver = all_solvers
+    benchmark(fastSNP, model)
+
+
+def test_find_active_reactions(model, all_solvers):
+    """Test find_active_reactions."""
+
+    model.solver = all_solvers
+    # solve LPs
+    active_rxns_lp = find_active_reactions(model)
+    # solving MILP or Fast-SNP may not be feasible for some solvers
+    # active_rxns_milp = find_active_reactions(model, solve="milp")
+    # active_rxns_fastsnp = find_active_reactions(model, solve="fastSNP")
+
+    active_rxns = ['ACALD', 'ACALDt', 'ACKr', 'ACONTa', 'ACONTb', 'ACt2r',
+                   'ADK1', 'AKGDH', 'AKGt2r', 'ALCD2x', 'ATPM', 'ATPS4r',
+                   'Biomass_Ecoli_core', 'CO2t', 'CS', 'CYTBD',
+                   'D_LACt2', 'ENO', 'ETOHt2r', 'EX_ac_e', 'EX_acald_e',
+                   'EX_akg_e', 'EX_co2_e', 'EX_etoh_e', 'EX_for_e',
+                   'EX_glc__D_e', 'EX_glu__L_e', 'EX_h_e', 'EX_h2o_e',
+                   'EX_lac__D_e', 'EX_nh4_e', 'EX_o2_e', 'EX_pi_e',
+                   'EX_pyr_e', 'EX_succ_e', 'FBA', 'FBP', 'FORt2', 'FORti',
+                   'FRD7', 'FUM', 'G6PDH2r', 'GAPD', 'GLCpts', 'GLNS',
+                   'GLUDy', 'GLUN', 'GLUSy', 'GLUt2r', 'GND', 'H2Ot',
+                   'ICDHyr', 'ICL', 'LDH_D', 'MALS', 'MDH', 'ME1', 'ME2',
+                   'NADH16', 'NADTRHD', 'NH4t', 'O2t', 'PDH', 'PFK', 'PFL',
+                   'PGI', 'PGK', 'PGL', 'PGM', 'PIt2r', 'PPC', 'PPCK', 'PPS',
+                   'PTAr', 'PYK', 'PYRt2', 'RPE', 'RPI', 'SUCCt2_2', 'SUCCt3',
+                   'SUCDi', 'SUCOAS', 'TALA', 'THD2', 'TKT1', 'TKT2', 'TPI']
+
+    assert set(active_rxns_lp) == set(active_rxns)
+    # assert set(active_rxns_milp) == set(active_rxns)
+    # assert set(active_rxns_fastsnp) == set(active_rxns)
+
+
+def test_find_reactions_in_cycles(large_model, all_solvers):
+    """Test find_reactions_in_cycles."""
+
+    large_model.solver = all_solvers
+    # solve LPs
+    rxns_in_cycles_lp = find_reactions_in_cycles(large_model)
+    # solving MILP or Fast-SNP may not be feasible for some solvers
```

Comment:
You can make tests conditional on the available solvers. We can help you with that or you could look through some of the other tests for some examples.

Reply:
Could you refer me to one of the examples? Have made other changes.